### PR TITLE
feat!: Added Address Descriptors to Geocoding response. Refactored Geo…

### LINKED ISSUE #953: Add support for Address Descriptors to Geocoding
Thanks for stopping by to let us know something could be better!

---
**PLEASE READ** 

If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/). This will ensure a timely response. 

Discover additional support services for the Google Maps Platform, including developer communities, technical guidance, and expert support at the Google Maps Platform [support resources page](https://developers.google.com/maps/support/). 

If your bug or feature request is not related to this particular library, please visit the Google Maps Platform [issue trackers](https://developers.google.com/maps/support/#issue_tracker).

Check for answers on StackOverflow with the [google-maps](http://stackoverflow.com/questions/tagged/google-maps) tag.

---

 **Describe the solution you'd like**
Add support for Address Descriptors in the geocoding request.

 **Additional context**
https://mapsplatform.google.com/demos/address-descriptors/

### PR DESCRIPTION
**Pull request by tcliff111**
…coding response to allow fields outside the GeocodingResult to be exposed through the client.

Thank you for opening a Pull Request!

---

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ X] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X ] Ensure the tests and linter pass
- [ X] Code coverage does not decrease (if any source code was changed)
- [X ] Appropriate docs were updated (if necessary)

Fixes #953 🦕

### CODE REVIEW

File: src/main/java/com/google/maps/GeocodingApiRequest.java [hunk 1]
```diff
@@ -20,13 +20,13 @@
 import com.google.maps.internal.ApiConfig;
 import com.google.maps.model.AddressType;
 import com.google.maps.model.ComponentFilter;
-import com.google.maps.model.GeocodingResult;
+import com.google.maps.model.GeocodingResponse;
 import com.google.maps.model.LatLng;
 import com.google.maps.model.LocationType;
 
 /** A request for the Geocoding API. */
```
Comment (wangela) [inline src/main/java/com/google/maps/GeocodingApiRequest.java:27]: Let's improve upon this javadoc since we're changing GeocodingApiRequest.

**Review by wangela — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/maps/GeocodingApi.java [hunk 2]
```diff
@@ -68,19 +70,23 @@ public static GeocodingApiRequest reverseGeocode(GeoApiContext context, LatLng l
     return request;
   }
 
-  public static class Response implements ApiResponse<GeocodingResult[]> {
+  public static class Response implements ApiResponse<GeocodingResponse> {
```
[the remark this answers is not in the document]
Comment (tcliff111) [reply src/main/java/com/google/maps/GeocodingApi.java:76]: I added Javadoc comments to all the new fields.

File: src/main/java/com/google/maps/GeocodingApiRequest.java (hunk 1 above)
Comment (tcliff111) [reply src/main/java/com/google/maps/GeocodingApiRequest.java:27]: I added Javadoc comments to all the new fields.

**Comment by tcliff111** [comment]
Added new comments to all the new fields, please take a look.

**Review by wangela — APPROVED** [review]

**Pull request merged**